File: example-java/app/src/main/java/com/vibes/androidsdkexampleapp/notification/MyNotificationFactory.java
package com.vibes.androidsdkexampleapp.notification;

import android.content.Context;
import androidx.core.app.NotificationCompat;

import com.vibes.vibes.NotificationFactory;
import com.vibes.vibes.PushPayloadParser;
import com.vibes.androidsdkexampleapp.R;

public class MyNotificationFactory extends NotificationFactory {
    public MyNotificationFactory(Context context) {
        super(context);
    }

    /**
     * Customize the notification built
     */
    @Override
    public NotificationCompat.Builder getBuilder(PushPayloadParser pushModel, Context context) {
        return super.getBuilder(pushModel, context)
                .setSmallIcon(R.drawable.firebase_icon);
    }

}

File: example-kotlin/app/src/main/java/com/android/exampleapp/kotlin/notification/MyNotificationFactory.kt
package com.android.exampleapp.kotlin.notification

import android.content.Context
import androidx.core.app.NotificationCompat
import com.vibes.vibes.NotificationFactory
import com.vibes.vibes.PushPayloadParser
import com.android.exampleapp.kotlin.R;

/**
 * A factory that builds notifications received from Vibes Push Service
 *
 * @author edem.morny@vibes.com
 */
class MyNotificationFactory : NotificationFactory {

    constructor(context: Context?) : super(context)

    override fun getBuilder(pushModel:PushPayloadParser, context:Context): NotificationCompat.Builder{
        return super.getBuilder(pushModel, context).setSmallIcon(R.drawable.firebase_icon)
    }
}